be/src/agent/heartbeat_server.cpp

```diff
@@ -122,6 +123,7 @@ void HeartbeatServer::heartbeat(THeartbeatResult& heartbeat_result, const TMaste
         } else {
             heartbeat_result.backend_info.__set_is_set_storage_path(false);
         }
+        heartbeat_result.backend_info.__set_num_tablets(g_worker->num_shards());
```

Comment:
This operation will try to hold a lock. I'm worried that it will block the leader if this RPC is stuck on this lock, which will make the whole cluster fail. So is there another way to make this operation lock-free?

Reply:
can remove the lock by an atomic counter, the counter is not necessary to be accurate.